InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,380 @@
+                key: nameof(Intern.LastName),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.Email),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.PhoneNumber),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.Status),
+                values: "Text is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.UpdatedDate),
+                values: new String[] { "Date is required", "Date is the same as CreatedDate" });
+
+            invalidInternException.AddData(
+                key: nameof(Intern.CreatedDate),
+                values: "Date is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.JoinDate),
+                values: "Date is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.UpdatedBy),
+                values: "Id is required");
+
+            invalidInternException.AddData(
+                key: nameof(Intern.CreatedBy),
+                values: "Id is required");
+
+            var expectedInternValidationException =
+                new InternValidationException(invalidInternException);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(invalidIntern);
+
+            InternValidationException actualInternValidationException =
+                await Assert.ThrowsAsync<InternValidationException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternValidationException.Should().BeEquivalentTo(
+                expectedInternValidationException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.UpdateInternAsync(It.IsAny<Intern>()),
+                    Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyIfCreatedAndUpdatedDatesAreSameAndLogItAsync()
+        {
+            // given
+            Intern randomIntern = CreateRandomIntern();
+            DateTimeOffset sameDate = randomIntern.CreatedDate;
+            Intern invalidIntern = randomIntern;
+            invalidIntern.UpdatedDate = sameDate;
```

Comment:
Just pass `randomDate` to CreateRandomIntern method and create model with same dates.